UIPickerExtensions.swift

```diff
@@ -0,0 +1,57 @@
+import UIKit
+import ViewModelsSample
+
+extension UIPickerView {
+    private struct AssociatedKeys {
+        static var elementsKey = UnsafeMutablePointer<Int8>.allocate(capacity: 1)
+    }
+
+    private var elements: [String]? {
```

Comment:
Should this be an empty array instead of nil? We could avoid doing optional checks further down.